packages/auth-client/src/index.ts

```diff
@@ -463,7 +475,11 @@ export class AuthClient {
     /**
      * Callback once login has completed
      */
-    onSuccess?: (() => void) | (() => Promise<void>);
+    onSuccess?:
+      | (() => void)
+      | ((message?: InternetIdentityAuthResponseSuccess) => void)
+      | (() => Promise<void>)
+      | ((message?: InternetIdentityAuthResponseSuccess) => Promise<void>);
```

Comment:
making the `message` optional means the user cannot actually rely on `message` existing I think! how about something like this?

`(message: IIAuthResponse) => void | Promise<void>`

that should represent the same type, though will prevent the user from performing an unnecessary undefined check

Reply:
```suggestion
      | (() => void | Promise<void>)
      | (message: IIAuthResponse) => void | Promise<void>);
```
I think that we can do that, and also offer the same version with no message, like this. It'll be half as verbose

Reply:
I had a test that failed because the signature changed, which signaled to me that supporting the zero argument signature is important for compatibility

Reply:
Ok, quite surprising! Won't this force the users to check for undefined even though the argument will never be undefined?